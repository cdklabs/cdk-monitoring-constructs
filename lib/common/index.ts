export * from "./alarm";
export * from "./metric";
export * from "./monitoring";
export * from "./url";
export * from "./widget";
export * from "./strings";
