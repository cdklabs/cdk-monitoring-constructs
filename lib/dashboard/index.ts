export * from "./BitmapDashboard";
export * from "./DashboardRenderingPreference";
export * from "./DynamicDashboardSegment";
export * from "./DashboardSegment";
export * from "./DashboardWithBitmapCopy";
export * from "./DefaultDashboardFactory";
export * from "./DynamicDashboardFactory";
export * from "./DefaultWidgetFactory";
export * from "./IDashboardFactory";
export * from "./IDynamicDashboardFactory";
export * from "./IWidgetFactory";
export * from "./MonitoringNamingStrategy";
export * from "./SingleWidgetDashboardSegment";
export * from "./widget";
