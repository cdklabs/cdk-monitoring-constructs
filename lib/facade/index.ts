export * from "./IMonitoringAspect";
export * from "./MonitoringFacade";
export * from "./MonitoringAspect";
