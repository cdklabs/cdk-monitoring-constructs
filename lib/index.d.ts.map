{"version":3,"file":"index.d.ts","sourceRoot":"","sources":["index.ts"],"names":[],"mappings":"AAAA,cAAc,UAAU,CAAC;AACzB,cAAc,aAAa,CAAC;AAC5B,cAAc,UAAU,CAAC;AACzB,cAAc,cAAc,CAAC"}