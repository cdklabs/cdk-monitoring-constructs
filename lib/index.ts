export * from "./common";
export * from "./dashboard";
export * from "./facade";
export * from "./monitoring";
