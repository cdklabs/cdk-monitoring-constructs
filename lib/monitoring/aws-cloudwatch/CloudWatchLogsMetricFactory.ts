import { DimensionsMap } from "aws-cdk-lib/aws-cloudwatch";

import {
  BaseMetricFactory,
  BaseMetricFactoryProps,
  MetricFactory,
  MetricStatistic,
} from "../../common";

const CloudWatchLogsNamespace = "AWS/Logs";

export interface CloudWatchLogsMetricFactoryProps
  extends BaseMetricFactoryProps {
  /**
   * Name of the log group to monitor.
   */
  readonly logGroupName: string;
}

export class CloudWatchLogsMetricFactory extends BaseMetricFactory<CloudWatchLogsMetricFactoryProps> {
  private readonly dimensionsMap: DimensionsMap;

  constructor(
    metricFactory: MetricFactory,
    props: CloudWatchLogsMetricFactoryProps,
  ) {
    super(metricFactory, props);

    this.dimensionsMap = {
      LogGroupName: props.logGroupName,
    };
  }

  metricIncomingLogEvents() {
    return this.metricFactory.createMetric(
      "IncomingLogEvents",
      MetricStatistic.N,
      "Logs",
      this.dimensionsMap,
      undefined,
      CloudWatchLogsNamespace,
      undefined,
      this.region,
      this.account,
    );
  }
}
