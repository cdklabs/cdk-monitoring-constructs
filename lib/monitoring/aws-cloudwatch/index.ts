export * from "./CloudWatchLogsMetricFactory";
export * from "./LogMonitoring";
