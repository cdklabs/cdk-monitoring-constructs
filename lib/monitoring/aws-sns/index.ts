export * from "./SnsTopicMetricFactory";
export * from "./SnsTopicMonitoring";
