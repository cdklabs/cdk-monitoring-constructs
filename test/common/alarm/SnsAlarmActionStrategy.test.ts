import { Stack } from "monocdk";
import { Template } from "monocdk/assertions";
import { Alarm, Metric } from "monocdk/aws-cloudwatch";
import { Topic } from "monocdk/aws-sns";

import { SnsAlarmActionStrategy } from "../../../lib";

test("snapshot test: default action only", () => {
  const stack = new Stack();
  const onAlarmTopic = new Topic(stack, "DummyTopic");
  const alarm = new Alarm(stack, "DummyAlarm", {
    evaluationPeriods: 1,
    threshold: 0,
    metric: new Metric({ namespace: "Dummy", metricName: "Dummy" }),
  });
  const action = new SnsAlarmActionStrategy({ onAlarmTopic });
  action.addAlarmActions({ alarm, action });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});

test("snapshot test: all actions", () => {
  const stack = new Stack();
  const onAlarmTopic = new Topic(stack, "DummyTopic");
  const onOkTopic = new Topic(stack, "DummyOkTopic");
  const onInsufficientDataTopic = new Topic(
    stack,
    "DummyInsufficientDataTopic"
  );
  const alarm = new Alarm(stack, "DummyAlarm", {
    evaluationPeriods: 1,
    threshold: 0,
    metric: new Metric({ namespace: "Dummy", metricName: "Dummy" }),
  });
  const action = new SnsAlarmActionStrategy({
    onAlarmTopic,
    onOkTopic,
    onInsufficientDataTopic,
  });
  action.addAlarmActions({ alarm, action });

  expect(Template.fromStack(stack)).toMatchSnapshot();
});
