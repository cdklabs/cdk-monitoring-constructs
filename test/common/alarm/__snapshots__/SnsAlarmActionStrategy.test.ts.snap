// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: all actions 1`] = `
Object {
  "Resources": Object {
    "DummyAlarm234203A9": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "DummyTopic75B5E033",
          },
        ],
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "EvaluationPeriods": 1,
        "InsufficientDataActions": Array [
          Object {
            "Ref": "DummyInsufficientDataTopic5C2BB5B5",
          },
        ],
        "MetricName": "Dummy",
        "Namespace": "Dummy",
        "OKActions": Array [
          Object {
            "Ref": "DummyOkTopicD5349398",
          },
        ],
        "Period": 300,
        "Statistic": "Average",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "DummyInsufficientDataTopic5C2BB5B5": Object {
      "Type": "AWS::SNS::Topic",
    },
    "DummyOkTopicD5349398": Object {
      "Type": "AWS::SNS::Topic",
    },
    "DummyTopic75B5E033": Object {
      "Type": "AWS::SNS::Topic",
    },
  },
}
`;

exports[`snapshot test: default action only 1`] = `
Object {
  "Resources": Object {
    "DummyAlarm234203A9": Object {
      "Properties": Object {
        "AlarmActions": Array [
          Object {
            "Ref": "DummyTopic75B5E033",
          },
        ],
        "ComparisonOperator": "GreaterThanOrEqualToThreshold",
        "EvaluationPeriods": 1,
        "MetricName": "Dummy",
        "Namespace": "Dummy",
        "Period": 300,
        "Statistic": "Average",
        "Threshold": 0,
      },
      "Type": "AWS::CloudWatch::Alarm",
    },
    "DummyTopic75B5E033": Object {
      "Type": "AWS::SNS::Topic",
    },
  },
}
`;
